source/type/pt_data.cpp

```diff
@@ -35,6 +35,98 @@ www.navitia.io
 
 namespace navitia { namespace type {
 
+type::Network* PT_Data::get_or_create_network(const std::string& uri, const std::string& name, int sort) {
+    const auto it = networks_map.find(uri);
+    if (it != networks_map.end()) {
+        return it->second;
+    }
+
+    nt::Network* network = new nt::Network();
+    network->uri = uri;
+    network->name = name;
+    network->sort = sort;
+
+    network->idx = networks.size();
+    networks.push_back(network);
+    networks_map[uri] = network;
+
+    return network;
+}
+
+type::CommercialMode* PT_Data::get_or_create_commercial_mode(const std::string& uri, const std::string& name) {
+    const auto it = commercial_modes_map.find(uri);
+    if (it != commercial_modes_map.end()) {
+        return it->second;
+    }
+
+    nt::CommercialMode* mode = new nt::CommercialMode();
+    mode->uri = uri;
+    mode->name = name;
```

Comment:
When you create a CommercialMode you need to fill
- std::vector<Line*> line_list; In the same way of Network

Reply:
okay that is here
https://github.com/pbougue/navitia/pull/2/files#diff-12123f76f97a0b21544ff33ec37d8ffaR95

Reply:
okay that's here 
https://github.com/pbougue/navitia/pull/2/files#diff-12123f76f97a0b21544ff33ec37d8ffaR95